### DISCUSSION TITLE
Accessing /my-collection/tracks & /my-collection/albums through the API

### DISCUSSION THREAD

**Discussion by duttonst** [tidal-music/discussions#90] [Q&A]
Hi.

Is there an endpoint where I can access the user's current lists of 'hearted' albums & tracks?

Thanks in advance.

**Comment by stefanmoro** [comment]
Hey @duttonst ,

We currently don't expose "My Collection" on the TIDAL API. It is planned, and we are currently targeting a roll out in Q4.

Thanks

**Reply by duttonst** [reply]
Thank you!

**Reply by Bowlerr** [reply]
@stefanmoro Any update on this? Wanted to be able to grab my liked artists from "My Collection"

**Reply by stefanmoro** [reply]
Hey @Bowlerr, it is in the works and we aim to roll out fully early next year.

**Reply by theikid** [reply]
Hi @stefanmoro any update on this ? Thanks !

**Reply by bglw** [reply]
Poking my nose into https://tidal-music.github.io/tidal-api-reference/ I notice their `Internal` tier has endpoints for `userCollections` which is new 🤞

**Comment by stefanmoro** [comment]
Hey,

As of a few months back we have started rolling out user collection access. It is currently possible to R/W albums, artists and playlists. We are working on the remaining functionality and will keep you posted.

**Reply by duttonst** [reply]
Playlists & Artists work Ok but Albums are not working for me:
<img width="1438" alt="Screenshot 2025-04-28 at 08 52 38" src="https://github.com/user-attachments/assets/736d92ed-4ae1-4896-a95d-a98f0fac7e77" />

**Reply by stefanmoro** [reply]
Thanks. Can you please try again?

**Reply by duttonst** [reply]
Looking better: I'm seeing data now. Many thanks.

**Comment by Remideza** [comment]
Hello, following back on your answer, I've accessed user collections, however the original question was about fetching liked tracks

I didn't find a workaround using userCollections for this
I tried via playlists, but the users liked tracks doesn't look like to be listed into playlists 

Is there a way or the includes=tracks is to be included later ?

**Comment by drestefan-cpu** [comment]
Hey @stefanmoro, wanted to check in on the userCollections rollout. I have collection.read enabled but both of these are returning 404 with a fresh token:
userCollections/{id}/relationships/artists
userCollections/{id}/relationships/albums

Anything needed on my end to get access?
thx!
